### DISCUSSION TITLE
Provisioning Script and building template vars from Grafana Orgs

### DISCUSSION THREAD

**Discussion by allomorphy** [grafana/grafana#43700] [Dashboards, Panels & Visualizations]
👋  Folks,

_If this message is not suitable for this forum, then please accept my apologies and delete_.

Is there a place of community contributed resources? I was thinking to place the following script where it might be of use to others. [https://github.com/DIFFERENCECODE/grafana-provision-user](https://github.com/DIFFERENCECODE/grafana-provision-user)

The script is used to automate the creation of new Organisations and Users in those Organisations.

My question concerns our requirement to define template variables for a dashboard dropdown to be populated by one or more users according to the following rules:

- If the logged in user has the Role of Editor, then we we want the variable to contain all the users in the organisation.
- If the logged in user has a Viewer Role, then only their name should be in the variable/dropdown.

I could not find an answer in the docs or on any forum, where it is it possible to build the query in the web interface for these variables from with the Grafana API or from the internal grafana.db? (I tried the -- Grafana -- datasource in the dashboard to define variables and alas no dice).
If we could use either of these to build these templated variables by defining them in the Grafana web interface, that would be great. (Ideally want to avoid development of a plugin for this purpose).

If posting does not give us a way forward a direct method, then we will use the above script to replicate/duplicate the Grafana instance Role information in influxdb. From influx, it should be straightforward to build a query according to the above rules (but an ugly solution that we wish to avoid).

Would love to receive some response 🤗 
Eric

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
